universal-login-commons/lib/core/utils/authorisation.ts

```diff
@@ -0,0 +1,24 @@
+import {utils} from 'ethers';
+
+import {CancelAuthorisationRequest} from '../models/authorisation';
+
+export const hashCancelAuthorisationRequest =
+  (cancelAuthorisationRequest: CancelAuthorisationRequest): string => {
+    const payloadString = JSON.stringify(cancelAuthorisationRequest);
+    const payloadBytes = utils.toUtf8Bytes(payloadString);
+    return utils.keccak256(payloadBytes);
+  };
+
+export const signCancelAuthorisationRequest =
+  (cancelAuthorisationRequest: CancelAuthorisationRequest, privateKey: string): utils.Signature => {
+    const payloadDigest = hashCancelAuthorisationRequest(cancelAuthorisationRequest);
+    const signingKey = new utils.SigningKey(privateKey);
```

Comment:
👍  Good job finding that low-level function, I was not aware it exists and created a Wallet each time.

Reply:
Other than that, there is lots of `const address = new Wallet(privateKey).address` lines.
Can be replaced with `const address = utils.computeAddress(privateKey)`

Reply:
[example 1](https://github.com/UniversalLogin/UniversalLoginSDK/blob/1fc6e900c05cc73eaa416ddde68cad75b9007855/universal-login-commons/lib/core/utils/messages/calculateMessageSignature.ts#L41)

[example 2](https://github.com/UniversalLogin/UniversalLoginSDK/blob/1fc6e900c05cc73eaa416ddde68cad75b9007855/universal-login-wallet/src/core/services/ConnectToWallet.ts#L8)

etc..